src/main/java/net/minecraftforge/common/extensions/IForgeFluid.java

```diff
@@ -19,103 +19,333 @@
+     * @return {@code null} for default behavior; otherwise, returns the fluid's {@link BlockPathTypes} for pathfinding purposes
      */
     @Nullable
-    default Boolean isAABBInsideMaterial(FluidState state, LevelReader world, BlockPos pos, AABB boundingBox, Material materialIn)
+    default BlockPathTypes getBlockPathType(FluidState state, BlockGetter level, BlockPos pos, @Nullable Mob entity)
     {
         return null;
     }
 
     /**
-     * Called when entities are moving to check if they are inside a liquid
-     *
-     * @param world world that is being tested.
-     * @param pos block thats being tested.
-     * @param boundingBox box to test, generally the bounds of an entity that are besting tested.
-     * @return null for default behavior, true if the box is within the material, false if it was not.
+     * Gets the {@link BlockPathTypes} of the fluid when adjacent to some pathfinding entity.
+     * <ul>
+     * <li>Negative Values = Untraversable</li>
+     * <li>0 = Best</li>
+     * <li>Highest = Worst</li>
+     * </ul>
+     * @param state the current FluidState
+     * @param level the current {@link Level}'s block reader
+     * @param pos the position of the fluid
+     * @param entity the pathing entity, can be null
+     * @param originalType the {@link BlockPathTypes} obtained from {@link IForgeBlock#getBlockPathType(BlockState, BlockGetter, BlockPos, Mob)}
+     * @return {@code null} for default behavior; otherwise, returns the fluid's adjacent {@link BlockPathTypes}
      */
     @Nullable
-    default Boolean isAABBInsideLiquid(FluidState state, LevelReader world, BlockPos pos, AABB boundingBox)
+    default BlockPathTypes getAdjacentBlockPathType(FluidState state, BlockGetter level, BlockPos pos, @Nullable Mob entity, BlockPathTypes originalType)
     {
         return null;
     }
 
     /**
-     * Location sensitive version of getExplosionResistance
+     * Handles acceleration or "pushing" while moving through the fluid.
+     * This implementation is slightly modified default behavior for fluid acceleration, based on {@link Entity#updateFluidHeightAndDoFluidPushing(net.minecraft.tags.Tag, double)}.
      *
-     * @param world The current world
-     * @param pos Block position in world
-     * @param explosion The explosion
-     * @return The amount of the explosion absorbed.
+     * @param state The {@link FluidState} the entity is in
+     * @param entity The current {@link Entity} that motion is being applied to
+     * @return Whether the motion was successfully applied to the {@link Entity}
      */
-    @SuppressWarnings("deprecation")
-    default float getExplosionResistance(FluidState state, BlockGetter world, BlockPos pos, Explosion explosion)
+    default boolean updateFluidHeightAndDoFluidPushing(FluidState state, Entity entity)
     {
-        return state.getExplosionResistance();
+        if (entity.touchingUnloadedChunk()) return false;
+        else
+        {
```

Comment:
Personally I would say remove this else statement as the other part of it always returns so there is no real need to have all the contents of this else indented one spot further